### PR TITLE
Add SPV_GOOGLE_hlsl_functionality1

### PR DESCRIPTION
**Pull request by antiagainst**
This extension introduces two new decorations to extend HLSL
functionality: HlslCounterBuffer and HlslSemantic.

### CODE REVIEW

File: extensions/GOOGLE/SPV_GOOGLE_hlsl_functionality1.asciidoc [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+2+^.^| Decoration | Enabling Capabilities 2+<.^| Extra Operands
+| 5634 | *HlslCounterBufferGOOGLE* +
+An <id> of the counter buffer associated with the target buffer.
+  |
+  2+| _<id>_ +
+      Counter Buffer ID
+| 5635 | *HlslSemanticGOOGLE* +
+A string describing the intended use of a value.
+Use this to express user-defined and system semantic values for shaders
+originally written in HLSL.
```
Comment (johnkslang) [inline extensions/GOOGLE/SPV_GOOGLE_hlsl_functionality1.asciidoc]: Say anything about case sensitivity? Or what case things should come out?

File: extensions/GOOGLE/SPV_GOOGLE_hlsl_functionality1.asciidoc [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+|========================================
+
+Dependencies
+------------
+
+This extension is written against the SPIR-V Specification,
+Version 1.2 Revision 4, Unified.
+
+This extension requires SPIR-V 1.2.
+
```
[the remark this answers is not in the document]
Comment (antiagainst) [reply extensions/GOOGLE/SPV_GOOGLE_hlsl_functionality1.asciidoc]: Yep. Good catch. Fixed.

I also changed "requires SPIR-V 1.2" to "requires SPIR-V 1.0".

File: extensions/GOOGLE/SPV_GOOGLE_hlsl_functionality1.asciidoc [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+Overview
+--------
+
+This extension provides two new decorations to extend HLSL functionality:
+HlslCounterBuffer and HlslSemantic.
+
+The decoration HlslCounterBuffer is used with OpDecorateId to link a counter
+buffer to a UAV resources that has an associated counter.
```
[the remark this answers is not in the document]
Comment (antiagainst) [reply extensions/GOOGLE/SPV_GOOGLE_hlsl_functionality1.asciidoc]: Done

File: extensions/GOOGLE/SPV_GOOGLE_hlsl_functionality1.asciidoc (hunk 1 above)
Comment (antiagainst) [reply extensions/GOOGLE/SPV_GOOGLE_hlsl_functionality1.asciidoc]: Added a sentence to say this is case insensitive, following HLSL tradition. Whether we convey the original semantic string or do normalization in the compilers are implementation specific; I feel we should not mandate that in the spec.

**Pull request merged**